File: backend-integration-test/src/test/java/hu/bhr/backend/customer/dto/ErrorResponse.java
package hu.bhr.backend.customer.dto;

import io.cucumber.core.internal.com.fasterxml.jackson.annotation.JsonProperty;

public record ErrorResponse(
        @JsonProperty("errorCode") String errorCode,
        @JsonProperty("errorMessage") String errorMessage,
        @JsonProperty("timestamp") String timestamp
) {}

File: crm-frontend/src/lib/models/error-response.ts
export type ErrorResponse = {
  errorCode: string;
  errorMessage: string;
  timestamp: string;
};

export function isErrorResponse(object: any): object is ErrorResponse {
  return (
    typeof object.errorMessage === "string" &&
    typeof object.errorCode === "string" &&
    typeof object.timestamp === "string"
  );
}
